### PR TITLE
refactor(core): use global API endpoint wherever possible

### PR DESCRIPTION
**Pull request by cngonzalez**
### Description
This PR replicates one made in the v3 branch to normalize all calls, when possible, to a `/projects/{projectId}/datasets/{dataset}` endpoint. This helps keep everything somewhat predictable and aligned.

The `scope` parameter is deprecated as a result (but should still be functional).

The users store was also slightly changed because a project hostname is not needed for `p-{user}` anymore.

### What to review
Most relevant changes are in the client store. Any odd edge cases? Anything I'm missing?

### Testing
A few tests were added or updated.

### CODE REVIEW

**Comment by cngonzalez** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/sanity-io/sdk/817?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#817** <a href="https://app.graphite.com/github/pr/sanity-io/sdk/817?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/sanity-io/sdk/817?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#816** <a href="https://app.graphite.com/github/pr/sanity-io/sdk/816?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#825](https://github.com/sanity-io/sdk/pull/825) <a href="https://app.graphite.com/github/pr/sanity-io/sdk/825?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#813** <a href="https://app.graphite.com/github/pr/sanity-io/sdk/813?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/core/src/client/clientStore.ts [hunk 1]
```diff
@@ -179,40 +191,47 @@ export const getClient = bindActionGlobally(
...
     const apiHost = options.apiHost ?? instance.config.auth?.apiHost ?? getStagingApiHost()
 
-    const effectiveOptions: ClientConfig & {apiVersion: string} = {
+    const effectiveOptions = {
       ...DEFAULT_CLIENT_CONFIG,
-      ...((options.scope === 'global' || !projectId || resource) && {useProjectHostname: false}),
+      // assume explicit projectId / dataset rather than resource means project-specific client
+      // useProjectHostname will also navigate to the correct endpoint with just a resource
+      ...((options.projectId && options.dataset) || scope === 'default'
+        ? {useProjectHostname: true}
```
Comment (joshuaellis) [inline packages/core/src/client/clientStore.ts:223]: i thought you could still use the global API to query datasets? is there a benefit to still use the project hostname?

**Review by joshuaellis — APPROVED** [review]
Nothing blocking, out to the global api 🥳

File: packages/core/src/client/clientStore.ts (hunk 1 above)
Comment (cngonzalez) [reply packages/core/src/client/clientStore.ts:223]: this is more if a user has explicitly provided both. `useDatasets` will pass a `projectId` but hits `/projects/{projectId}/datasets`
